Providers/Scripts/2.4x-2.5x/Scripts/nxOMSSyslog.py

```diff
@@ -210,62 +285,219 @@ def ReadSyslogNGConf(SyslogSource):
     for d in SyslogSource:
-        if 'Severities' not in d.keys() or d['Severities'] is None or len(d['Severities']) is 0:
-            facility_txt = ''
-        else:
-            facility_txt = '\n#OMS_facility = ' + d['Facility'] + '\n'
+        if ('Severities' in d.keys() and d['Severities'] is not None
+                and len(d['Severities']) > 0):
+            facility_txt = '\n' + wkspc_comment + ' Facility = ' \
+                           + d['Facility'] + '\n'
             sevs = reduce(lambda x, y: x + ',' + y, d['Severities'])
-            facility_txt += 'filter f_' + \
-                d['Facility'] + \
-                '_oms { level(' + sevs + ') and facility(' + d[
-                    'Facility'] + '); };\n'
-            facility_txt += 'log { source(' + source_expr + '); filter(f_' + d[
-                'Facility'] + '_oms); destination(d_oms); };\n'
+            filter_str = 'f_' + d['Facility'] + '_' + WorkspaceID + '_oms'
+            facility_txt += 'filter ' + filter_str + ' { level(' + sevs \
+                            + ') and facility(' + d['Facility'] + '); };\n'
+            facility_txt += 'log { source(' + source_expr + '); filter(' \
+                            + filter_str + '); destination(' \
+                            + destination_str + '); };\n'
             txt += facility_txt
+
+    # Write the new complete txt to the conf file
     try:
         codecs.open(conf_path, 'w', 'utf8').write(txt)
-        LG().Log(
-            'INFO', 'Created omsagent syslog-ng configuration at ' + conf_path + '.')
+        LG().Log('INFO', 'Created omsagent syslog-ng configuration at ' + \
+                         conf_path + '.')
     except:
-        LG().Log(
-            'ERROR', 'Unable to create omsagent syslog-ng configuration at ' + conf_path + '.')
+        LG().Log('ERROR', 'Unable to create omsagent syslog-ng configuration ' \
+                          'at ' + conf_path + '.')
         return False
-    if os.system('sudo /opt/microsoft/omsconfig/Scripts/OMSSyslog-ng.post.sh') == 0:
+
+    if os.system('sudo /opt/microsoft/omsconfig/Scripts/' \
+                 'OMSSyslog-ng.post.sh') is 0:
         LG().Log('INFO', 'Successfully executed OMSSyslog-ng.post.sh.')
     else:
         LG().Log('ERROR', 'Error executing OMSSyslog-ng.post.sh.')
         return False
+
     return True
+
+
+def GetSyslogConfMultiHomedHeaderString(WorkspaceID):
+    """
+    Return the header for the multi-homed section from an rsyslog conf file
+    """
+    return '# OMS Syslog collection for workspace ' + WorkspaceID
+
+
+def ExtractPortFromFluentDConf(WorkspaceID):
+    """
+    Returns the port used for this workspace's syslog collection from the
+    FluentD configuration file:
+    If multi-homed:
+    /etc/opt/microsoft/omsagent/<workspace-ID>/conf/omsagent.d/syslog.conf
+    if not multi-homed:
+    /etc/opt/microsoft/omsagent/conf/omsagent.conf
+    """
+    omsagent_dir = '/etc/opt/microsoft/omsagent/'
```

Comment:
Can you move this path to the top of the file in constants section just like it is done for syslog paths and port?